### PR TITLE
dont apply drag methods if not draggable

### PR DESCRIPTION
**Pull request by krizka**
If not, dragging prevents normal working of custom drag logic, applied to tree leafs.

### CODE REVIEW

**Comment by zombieJ** [comment]
close #164

**Comment by zombieJ** [comment]
ref: https://github.com/ant-design/ant-design/issues/10859

File: src/TreeNode.jsx [hunk 1]
```diff
@@ -511,7 +512,7 @@ class TreeNode extends React.Component {
         onContextMenu={this.onContextMenu}
         onClick={this.onSelectorClick}
         onDoubleClick={this.onSelectorDoubleClick}
-        onDragStart={this.onDragStart}
+        onDragStart={drag && this.onDragStart}
```
[the remark this answers is not in the document]
Comment (krizka) [reply src/TreeNode.jsx]: In this case, `drag` could be `false` or any other falsey value, which will be passed to `onDragStart` and other methods, only `undefined` is safe here IMO

File: src/TreeNode.jsx [hunk 2]
```diff
@@ -578,7 +578,12 @@ class TreeNode extends React.Component {
       draggable,
     } } = this.context;
     const disabled = this.isDisabled();
-    const drag = draggable || undefined;
+    const dataOrAriaAttributeProps = Object.keys(otherProps).reduce((prev, key) => {
+      if ((key.substr(0, 5) === 'data-' || key.substr(0, 5) === 'aria-')) {
```
Comment (krizka) [inline src/TreeNode.jsx]: better use `key.startsWith('data-')` here

File: src/TreeNode.jsx [hunk 3]
```diff
@@ -600,11 +601,11 @@ class TreeNode extends React.Component {
...
 
-        onDragEnter={this.onDragEnter}
-        onDragOver={this.onDragOver}
-        onDragLeave={this.onDragLeave}
-        onDrop={this.onDrop}
-        onDragEnd={this.onDragEnd}
+        onDragEnter={draggable && this.onDragEnter}
+        onDragOver={draggable && this.onDragOver}
+        onDragLeave={draggable && this.onDragLeave}
+        onDrop={draggable && this.onDrop}
```
Comment (DavraYoung) [inline src/TreeNode.jsx:607]: Why do we prevent dropping by draggable?

Comment (DavraYoung) [reply src/TreeNode.jsx:607]: Documentation states: "whether can drag treeNode. ", nothing to do with drop functionality, so why do we hide it?

**Pull request merged**